aioaws/core.py

```diff
@@ -104,32 +105,39 @@ def add_signed_download_params(self, method: Literal['GET', 'POST'], url: URL, e
         assert expires >= 1, f'expires must be greater than or equal to 1, not {expires}'
         assert expires <= 604800, f'expires must be less than or equal to 604800, not {expires}'
         now = utcnow()
-        url = url.copy_merge_params(
-            {
-                'X-Amz-Algorithm': _AUTH_ALGORITHM,
-                'X-Amz-Credential': self._aws4_credential(now),
-                'X-Amz-Date': _aws4_x_amz_date(now),
-                'X-Amz-Expires': str(expires),
-                'X-Amz-SignedHeaders': 'host',
-            }
-        )
+        params = {
+            'X-Amz-Algorithm': _AUTH_ALGORITHM,
+            'X-Amz-Credential': self._aws4_credential(now),
+            'X-Amz-Date': _aws4_x_amz_date(now),
+            'X-Amz-Expires': str(expires),
+        }
+        if self.aws_session_token:
+            params.update({'X-Amz-Security-Token': self.aws_session_token})
+        params.update({'X-Amz-SignedHeaders': 'host'})
+        url = url.copy_merge_params(params)
         _, signature = self._aws4_signature(now, method, url, {'host': self.host}, 'UNSIGNED-PAYLOAD')
         return url.copy_add_param('X-Amz-Signature', signature)
 
     def upload_extra_conditions(self, dt: datetime) -> List[Dict[str, str]]:
-        return [
+        conditions = [
             {'x-amz-credential': self._aws4_credential(dt)},
             {'x-amz-algorithm': _AUTH_ALGORITHM},
             {'x-amz-date': _aws4_x_amz_date(dt)},
         ]
+        if self.aws_session_token:
+            conditions.append({'x-amz-security-token': self.aws_session_token})
+        return conditions
 
     def signed_upload_fields(self, dt: datetime, string_to_sign: str) -> Dict[str, str]:
-        return {
+        fields = {
             'X-Amz-Algorithm': _AUTH_ALGORITHM,
             'X-Amz-Credential': self._aws4_credential(dt),
             'X-Amz-Date': _aws4_x_amz_date(dt),
-            'X-Amz-Signature': self._aws4_sign_string(string_to_sign, dt),
         }
+        if self.aws_session_token:
+            fields.update({'X-Amz-Security-Token': self.aws_session_token})
```

Comment:
```suggestion
            fields['X-Amz-Security-Token'] = self.aws_session_token
```

Reply:
Yep, should have done this in the first place. Thank you. -> fb8f459